File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0);cout.tie(0);ios::sync_with_stdio(false);
    int n; cin >> n;
    vector<int> a(n);
    for(int i = 0; i < n; i++) cin >> a[i];
    vector<int> cnt(10, 0);
    int l, t, maxl; l = t = maxl = 0;
    for(int r = 0; r < n; r++){
        cnt[a[r]]++;
        if(cnt[a[r]] == 1) t++;
        while(t > 2){
            cnt[a[l]]--;
            if(cnt[a[l]] == 0) t--;
            l++;
        }
        maxl = max(maxl, r-l+1);
    }
    cout << maxl;
    return 0;
}

File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.py
import sys
print = lambda *args, sep=" ", end="\n": sys.stdout.write(sep.join(map(str, args)) + end)
input = lambda: sys.stdin.readline().rstrip('\r\n')

def main():
    n = int(input())
    a = tuple(map(int, input().split()))
    cnt = [0]*10
    l = t = maxl= 0
    for r in range(n):
        cnt[a[r]] += 1
        if cnt[a[r]] == 1: t += 1
        while t > 2:
            cnt[a[l]] -= 1
            if cnt[a[l]] == 0: t -= 1
            l += 1
        maxl = max(maxl, r-l+1)
    print(maxl)
    return 0
if __name__ == '__main__':
    sys.exit(main())